azclishell/az_completer.py

```diff
@@ -125,18 +125,18 @@ def get_completions(self, document, complete_event):
         self._is_command = True
 
         text = reformat_cmd(text)
-
         if text.split():
-            for comp in self.gen_cmd_and_param_completions(text):
+            for comp in sorted(
+                    list(self.gen_cmd_and_param_completions(text)), key=lambda a: a.text):
                 yield comp
 
-        for cmd in self.gen_cmd_completions(text):
+        for cmd in sorted(list(self.gen_cmd_completions(text)), key=lambda a: a.text):
             yield cmd
 
-        for val in self.gen_dynamic_completions(text):
+        for val in sorted(list(self.gen_dynamic_completions(text)), key=lambda a: a.text,):
             yield val
 
-        for param in self.gen_global_param_completions(text):
+        for param in sorted(list(self.gen_global_param_completions(text)), key=lambda a: a.text,):
```

Comment:
This should be in a helper method that you call.
Something like `sorted_completions(...)`